### ISSUE TITLE
Custom types in ArrayOf don't work

### ISSUE THREAD

**Issue by jefferai** [goadesign/goa#2472]
Goa 3.0.9

When I create a Type exactly as specified in the docs example:
```
var Hostname = Type("Hostname", String, func() {
    Description("A host name")
    Format(FormatHostname)
})
```
And then use it in ArrayOf, like so:
`Field(2, "workers", ArrayOf(Hostname), "", func() {...`
Then I get this error:
```
exit status 1
v is a string but res type is HostnameResponseBody
v is a string but res type is Hostname
failed to run protoc: exit status 1: clientapi_v1.proto:42:1: Expected "{".
```

**Comment by raphael** [comment]
https://github.com/goadesign/goa/pull/2272 makes it possible to define user types that are not objects. This would fix the above. This PR needs to be updated though before it can be merged. @nitinmohan87  is that something you'd be interested in doing? Thank you!

**Comment by nitinmohan87** [comment]
I will try to fix the PR for primitive user type this week and let you know.

**Comment by jefferai** [comment]
Comment to keep from staleness

**Comment by raphael** [comment]
The PR above has been merged in 3.1 and the design described in the issue should now work properly. @jefferai are you able to confirm? if so can you close this issue? thank you!

**Comment by jefferai** [comment]
I'm not able to confirm as we're no longer using Goa  :-(  Sorry -- just too many tradeoffs for our particular needs at the moment.

To be clear, I think it's a _really cool_ project but the amount of time we were spending on workarounds was more than we would spend just doing the basic coding ourselves.
